src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/RectangularArrayRecord.cs

```diff
@@ -155,8 +146,11 @@ internal static RectangularArrayRecord Create(BinaryReader reader, ArrayInfo arr
             BinaryType.Primitive => MapPrimitive((PrimitiveType)memberTypeInfo.Infos[0].AdditionalInfo!),
             BinaryType.PrimitiveArray => MapPrimitiveArray((PrimitiveType)memberTypeInfo.Infos[0].AdditionalInfo!),
             BinaryType.String => typeof(string),
+            BinaryType.StringArray => typeof(string[]),
             BinaryType.Object => typeof(object),
-            _ => typeof(ClassRecord)
+            BinaryType.ObjectArray => typeof(object[]),
+            BinaryType.Class or BinaryType.SystemClass => typeof(ClassRecord),
```

Comment:
same as below. Moreover, if this logic will change with https://github.com/dotnet/runtime/pull/107627, but I would prefer to merge this PR first (it's simpler and has no breaking changes)